File: bionic-js-tool/testing-code/swift/project-with-host-files/HostProject/host/libs/Vehicle.swift
import JavaScriptCore
import Bjs

class Vehicle: BjsObject {
    
    var description:String? {
        get {
            return Vehicle.bjs.getString(bjsGetProperty("description"))
        }
    }
    
    var weight:Double? {
        get {
            return Vehicle.bjs.getFloat(bjsGetProperty("weight"))
        }
        set {
            bjsSetProperty("weight", Vehicle.bjs.putPrimitive(newValue))
        }
    }
    
    var seats:Int? {
        get {
            return Vehicle.bjs.getInt(bjsGetProperty("seats"))
        }
    }
    
    var maxSpeed:Int? {
        get {
            return Vehicle.bjs.getInt(bjsGetProperty("maxSpeed"))
        }
    }
    
    private static var _bjsLocator: BjsLocator = BjsLocator("BeautifulVehicles", "Vehicle")
    override class var bjsLocator: BjsLocator { _bjsLocator }
    class func bjsFactory(_ jsObject: JSValue) -> Vehicle { Vehicle(jsObject) }
}

File: bionic-js-tool/testing-code/java/project-with-host-files/HostProject/src/main/java/test/project/host/libs/Vehicle.java
package test.project.host.libs;

import jjbridge.api.runtime.JSReference;
import jjbridge.api.value.strategy.FunctionCallback;
import bionic.js.Bjs;
import bionic.js.BjsTypeInfo;
import bionic.js.BjsObjectTypeInfo;
import bionic.js.Lambda;
import java.util.Date;
import bionic.js.BjsObject;

@BjsTypeInfo.BjsLocation(project = "BeautifulVehicles", module = "Vehicle")
public class Vehicle extends BjsObject {
    
    protected <T extends BjsObject> Vehicle(Class<T> type, JSReference jsObject) {
        super(type, jsObject);
    }
    
    protected <T extends BjsObject> Vehicle(Class<T> type, JSReference[] arguments) {
        super(type, arguments);
    }
    
    public Vehicle(JSReference jsObject) {
        this(Vehicle.class, jsObject);
    }
    
    public String description() {
        return bjs.getString(bjsGetProperty("description"));
    }
    
    public Double weight() {
        return bjs.getDouble(bjsGetProperty("weight"));
    }
    public void weight(Double value) {
        bjsSetProperty("weight", bjs.putPrimitive(value));
    }
    
    public Integer seats() {
        return bjs.getInteger(bjsGetProperty("seats"));
    }
    
    public Integer maxSpeed() {
        return bjs.getInteger(bjsGetProperty("maxSpeed"));
    }
    
    private static final JSReference bjsClass = BjsObjectTypeInfo.get(Vehicle.class).bjsClass();
    public static final Bjs bjs = BjsObjectTypeInfo.get(Vehicle.class).bjsLocator.get();
    public static final Bjs.JSReferenceConverter<Vehicle> bjsFactory = Vehicle::new;
}